Packs/SailPointIdentityNow/Integrations/SailPointIdentityNowEventCollector/SailPointIdentityNowEventCollector.py

```diff
@@ -0,0 +1,375 @@
+            prev_id: The id of the last event fetched
+        Returns:
+            List of events
+        """
+        if not prev_id:
+            query = {"indices": ["events"],
+                     "queryType": "SAILPOINT",
+                     "queryVersion": "5.2",
+                     "query":
+                     {"query": f"type:* AND created: [{from_date} TO now]"},
+                     "timeZone": "GMT",
+                     "sort": ["+created"],
+                     }
+        else:
+            query = {"indices": ["events"],
+                     "queryType": "SAILPOINT",
+                     "queryVersion": "5.2",
+                     "query":
+                     {"query": "type:* "},
+                     "sort": ["+id"],
+                     "searchAfter": [prev_id]
+                     }
+
+        url_suffix = f'/v3/search?limit={limit}'
+        demisto.debug(f'Searching for events with query: {query}.')
+        return self._http_request(method='POST', headers=self.headers, url_suffix=url_suffix, data=json.dumps(query))
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests API connectivity and authentication
+    Args:
+        client: Client object with the API client
+    Returns:
+        'ok' if test passed, anything else will fail the test
+    """
+
+    try:
+        fetch_events(
+            client=client,
+            limit=1,
+            last_run={},
+        )
+
+    except Exception as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+
+    return 'ok'
+
+
+def get_events(client: Client, from_date: str, from_id: str | None, limit: int = 50) -> tuple[List[Dict], CommandResults]:
+    """
+    Gets events from the SailPoint IdentityNow API
+    Args:
+        client: Client object with the API client
+        limit: Maximum number of events to fetch
+        from_date: The date from which to get events
+        from_id: The ID of an event from which to start to get events from
+    Returns:
+        List of events and CommandResults object
+    """
+    events = client.search_events(
+        prev_id=from_id,
+        from_date=from_date,
+        limit=limit
+    )
+    demisto.debug(f'Got {len(events)} events.')
+    hr = tableToMarkdown(name='Test Events', t=events)
+    return events, CommandResults(readable_output=hr)
+
+
+def fetch_events(client: Client,
+                 limit: int, last_run: dict[str, str]) -> tuple[Dict, List[Dict]]:
+    """
+    Fetches events from the SailPoint IdentityNow API
+    Args:
+        client: Client object with the API client
+        last_run: Dict containing the last run data
+        limit: Maximum number of events to fetch per call
+    Returns:
+        Tuple with the next run data and the list of events fetched
+    """
+    demisto.debug(f'Starting fetch_events with last_run: {last_run}.')
+    last_fetched_id = last_run.get('prev_id')
```

Comment:
```suggestion
    # last_fetched_id = last_run.get('prev_id')
```
Do we need it when running by time?